controllers/datadogagent/merger/secret.go

```diff
@@ -11,12 +11,14 @@ import (
 	corev1 "k8s.io/api/core/v1"
 
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/dependencies"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object"
 	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+	"github.com/go-logr/logr"
 )
 
 // SecretManager Kubernetes Secret Manager interface
 type SecretManager interface {
-	AddSecret(secretNamespace, secretName, key, value string) error
+	AddSecret(logger logr.Logger, secretNamespace, secretName, key, value string, extraMetadata map[string]string) error
```

Comment:
`extraMetadata` seems to only work for adding annotations so it can be name: `extraAnnotations`